backend/api/notifications/resources.py

```diff
@@ -1,241 +1,268 @@
-              type: string
-              default: Token sessionTokenHere==
-        responses:
-            200:
-                description: Message info
-            500:
-                description: Internal Server Error
-        """
-        unread_count = MessageService.has_user_new_messages(token_auth.current_user())
-        return unread_count, 200
+@router.get("/")
+async def get(
+    request: Request,
+    db: Database = Depends(get_db),
+    user: AuthUserDTO = Depends(login_required),
+):
+    """
+    Get all messages for logged in user
+    ---
+    tags:
+      - notifications
+    produces:
+      - application/json
+    parameters:
+        - in: header
+          name: Authorization
+          description: Base64 encoded session token
+          required: true
+          type: string
+          default: Token sessionTokenHere==
+        - in: query
+          name: messageType
+          type: string
+          description: Optional message-type filter; leave blank to retrieve all\n
+            Accepted values are 1 (System), 2 (Broadcast), 3 (Mention), 4 (Validation),
+            5 (Invalidation), 6 (Request team), \n
+            7 (Invitation), 8 (Task comment), 9 (Project chat),
+            10 (Project Activity), and 11 (Team broadcast)
+        - in: query
+          name: from
+          description: Optional from username filter
+          type: string
+        - in: query
+          name: project
+          description: Optional project filter
+          type: string
+        - in: query
+          name: taskId
+          description: Optional task filter
+          type: integer
+        - in: query
+          name: status
+          description: Optional status filter (read or unread)
+          type: string
+        - in: query
+          name: sortBy
+          description:
+            field to sort by, defaults to 'date'. Other useful options are 'read', 'project_id' and 'message_type'
+          type: string
+        - in: query
+          name: sortDirection
+          description: sorting direction ('asc' or 'desc'), defaults to 'desc'
+          type: string
+        - in: query
+          name: page
+          description: Page of results
+          type: integer
+        - in: query
+          name: pageSize
+          description: Size of page, defaults to 10
+          type: integer
+    responses:
+        200:
+            description: Messages found
+        404:
+            description: User has no messages
+        500:
+            description: Internal Server Error
+    """
+    preferred_locale = request.headers.get("accept-language")
+    page = request.query_params.get("page", 1)
+    page_size = request.query_params.get("pageSize", 10)
+    sort_by = request.query_params.get("sortBy", "date")
+    sort_direction = request.query_params.get("sortDirection", "desc")
+    message_type = request.query_params.get("messageType", None)
+    from_username = request.query_params.get("from")
```

Comment:
This is an interesting approach for endpoint query params. It should work, but typically you would define query params as function arguments and they would be parsed as such. If there is confusion, it's possible to explicitly define arguments as fastapi.Query